src/hooks/useEvent/useEvent.test.tsx

```diff
@@ -0,0 +1,41 @@
+import { renderHook } from '@testing-library/react-hooks'
+
+import { useEvent } from './useEvent'
+
+describe('useEvent', () => {
+  it('should call the callback provided when current result of the render is called', () => {
+    const callback = jest.fn()
+    const { result } = renderHook(value => useEvent(() => callback(value)), {
+      initialProps: { value: '1' },
+    })
+    expect(callback).toHaveBeenCalledTimes(0)
+
+    result.current()
+    expect(callback).toHaveBeenCalledTimes(1)
+    expect(callback).toHaveBeenCalledWith({ value: '1' })
+  })
+
+  it('should not fire callback when callback changes', () => {
```

Comment:
More than checking whether or not the callback is fired, can we have a test to check that the identity of the event handler is referentially stable (with `toEqual`)? This is the main purpose of the RFC, to avoid re-renders.

Reply:
Thank you for your feedback, I agree with you but I struggle to test this effectively.
Should I verify that `result.current` after re-rendering is equal to `result.current` before re-rendering ?

Reply:
Added a new unit test to check the stability of the event handler identity